senders/victorops/api/alert.go

```diff
@@ -0,0 +1,63 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// CreateAlertRequest the API request to be made to
+// create a victorops alert
+type CreateAlertRequest struct {
+	MessageType       MessageType `json:"message_type,omitempty"`
+	EntityID          string      `json:"entity_id,omitempty"`
+	EntityDisplayName string      `json:"entity_display_name,omitempty"`
+	StateMessage      string      `json:"state_message,omitempty"`
+	StateStartTime    int64       `json:"state_start_time,omitempty"`
+}
+
+// MessageType is the type of a victorops alert
+type MessageType string
+
+// Various possible MessageTypes
+const (
+	Critical        MessageType = "CRITICAL"
+	Warning         MessageType = "WARNING"
+	Acknowledgement MessageType = "ACKNOWLEDGEMENT"
+	Info            MessageType = "INFO"
+	Recovery        MessageType = "RECOVERY"
+)
+
+// CreateAlert creates a new alert in the victorops timeline
+func (client *Client) CreateAlert(routingKey string, alert CreateAlertRequest) error {
+	if alert.MessageType == "" {
+		return fmt.Errorf("field MessageType cannot be empty")
+	}
+
+	body := new(bytes.Buffer)
+	err := json.NewEncoder(body).Encode(alert)
+	if err != nil {
+		return fmt.Errorf("error while encoding json: %s", err)
+	}
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/%s", client.routingURL, routingKey), body)
```

Comment:
As I understand it in url there are three parts.
First: the constant https://alert.victorops.com/integrations/generic/20131114/alert.
Second: API key that you need to pass in the sender settings
Third: routingKey that the user passes

Let's get this to look like this?

Reply:
I am not sure whether the URL is a constant or unique to the user. That is why I designed it in such a way that the user will pass the entire URL and the api key via the sender settings, ie, https://alert.victorops.com/integrations/generic/20131114/alert/<api_key>. This is called the routingURL in the code. Thee routing key is then obtained from the web UI.

Reply:
Well, even if one of us is wrong, I do not see big problems in this decision =)